adapter/key_manager.go

```diff
@@ -92,8 +93,8 @@ func (adapter daoAdapter) Insert(name, password string) (string, string, error)
 }
 
 func (adapter daoAdapter) Recover(name, password, mnemonic string) (string, error) {
-	store, err := adapter.keyDAO.Read(name)
-	if err == nil {
+	store, err := adapter.keyDAO.Read(name, password)
```

Comment:
Do we need a new interface to check the key name existence？

Reply:
After `keyDAO` redesign, unified modification